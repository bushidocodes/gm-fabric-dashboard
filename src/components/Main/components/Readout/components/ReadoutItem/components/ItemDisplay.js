import styled, { css } from "styled-components";
import { PropTypes } from "prop-types";

import { spacingScale } from "style/styleFunctions";

// -webkit-flex: 1 1 0; added as a fix for flex-item not centering in safari 10.1
// for "cacheCard", set minimum height to overrule media query for  child readoutItemContainer size.  flex-basis of 100% is overruled in the parent container.
const ItemDisplay = styled.div`
  position: relative;
  padding: ${spacingScale(1)};

  button {
    align-self: center;
    margin: ${spacingScale(1)} 0;
  }

  color: ${props => props.color};
  ${props =>
    props.flex &&
    css`
      flex: ${props => props.flex};
    `};
`;

ItemDisplay.propTypes = {
  color: PropTypes.string,
  flex: PropTypes.string
};

export default ItemDisplay;
