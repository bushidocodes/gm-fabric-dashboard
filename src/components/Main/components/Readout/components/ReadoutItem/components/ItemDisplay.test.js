import React from "react";
import { shallow } from "enzyme";
import ItemDisplay from "./ItemDisplay";

describe("ItemDisplay", () => {
  it("matches snapshot", () => {
    const aItemDisplay = shallow(<ItemDisplay />);
    expect(aItemDisplay).toMatchSnapshot();
  });
  it("allows flex value to be overridden", () => {
    const aItemDisplay = shallow(<ItemDisplay flex="0 0 100%" />);
    expect(aItemDisplay).toHaveStyleRule("flex", "0 0 100%");
  });
});
