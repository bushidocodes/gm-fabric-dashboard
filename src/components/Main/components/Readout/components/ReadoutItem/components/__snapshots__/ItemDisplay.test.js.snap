// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDisplay matches snapshot 1`] = `
.c0 {
  position: relative;
  padding: 8px;
}

.c0 button {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 8px 0;
}

<div
  className="c0"
/>
`;
